fixit/cli/add_new_rule.py

```diff
@@ -86,32 +87,48 @@ def create_rule_file(file_path: Path, rule_name: str) -> None:
     print(f"Successfully created {file_path.name} rule file at {file_path.parent}")
 
 
-def main() -> None:
-    parser = argparse.ArgumentParser(
-        description="Creates a skeleton of adding new rule file"
-    )
+def _add_arguments(
+    parser: Union[argparse._SubParsersAction, argparse.ArgumentParser]
+) -> None:
+    """All required arguments for `add_new_rule` """
     parser.add_argument(
-        "-p",
         "--path",
         type=is_path_exists,
         default=Path("fixit/rules/new.py"),
         help="Path to add rule file, defaults to fixit/rules/new.py",
     )
-
     parser.add_argument(
-        "-n",
         "--name",
         type=str,
         default="",
         help="Name of the rule, defaults to `New`",
     )
 
-    args = parser.parse_args()
 
+def register_subparser(parser: argparse._SubParsersAction = None) -> None:
```

Comment:
The type is wrong.
It should be `Optional[argparse._SubParsersAction]`.

Reply:
Done